hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -355,6 +355,117 @@ def hpat_pandas_series_values_impl(self):
+        raise TypingError('{} The object must be boolean. Given sort: {}'.format(_func_name, sort))
+
+    if not isinstance(ascending, (types.Omitted, types.Boolean, bool)):
+        raise TypingError('{} The object must be boolean. Given ascending: {}'.format(_func_name, ascending))
+
+    if not isinstance(bins, (types.Omitted, types.NoneType)) and bins is not None:
+        raise TypingError('{} Unsupported parameters. Given bins: {}'.format(_func_name, bins))
+
+    if not isinstance(dropna, (types.Omitted, types.Boolean, bool)):
+        raise TypingError('{} The object must be boolean. Given dropna:{}'.format(_func_name, dropna))
+
+    if isinstance(self.data, StringArrayType):
+        def hpat_pandas_series_value_counts_str_impl(self, normalize=False, sort=True, ascending=False, bins=None, dropna=True):
+            # TODO: if dropna add nan handling
+
+            value_counts_dict = {}
+
+            for value in self._data:
+                if value in value_counts_dict:
+                    value_counts_dict[value] += 1
+                else:
+                    value_counts_dict[value] = 1
+
+            # TODO: workaround, keys() result can not be casted to array type
+            # TODO: use list comprehension instead or self.unique()?
+            unique_values = [key for key in value_counts_dict]
+            unique_values_len = len(unique_values)
+
+            value_counts = numpy.empty(unique_values_len, dtype=numpy.intp)
+            for i, key in enumerate(value_counts_dict):
+                value_counts[i] = value_counts_dict[key]
+
+            # Take initial order as default
+            indexes_order = numpy.arange(unique_values_len)
+            if sort:
+                indexes_order = numpy.array(value_counts).argsort()
+                if not ascending:
+                    indexes_order = indexes_order[::-1]
+
+            sorted_unique_values = [unique_values[i] for i in indexes_order]
+            sorted_value_counts = numpy.take(value_counts, indexes_order)
+
+            return pandas.Series(sorted_value_counts, index=sorted_unique_values)
+
+        return hpat_pandas_series_value_counts_str_impl
+
+    def hpat_pandas_series_value_counts_number_impl(self, normalize=False, sort=True, ascending=False, bins=None, dropna=True):
+        unique_values = self.unique()
+
+        if dropna:
+            nan_mask = numpy.isnan(unique_values)
+            unique_values = unique_values[~nan_mask]
+        #else:
+        # TODO: unique() can not handle numpy.nan because numpy.nan == numpy.nan is False
+
+        # TODO: not optimal
+        value_counts = numpy.array([numpy.sum(self._data == value) for value in unique_values])
+
+        # Series.unique() returns values in ascending order
+        indexes_order = numpy.arange(len(unique_values))
+        if sort:
+            indexes_order = numpy.array(value_counts).argsort()
+            if not ascending:
+                indexes_order = indexes_order[::-1]
```

Comment:
It would be better to check the line on max and min integer input data.

Reply:
One correction: let's check the line on **duplicated** min and max integer input data.

Reply:
I'm not sure it is necessary here. Could you give me any test case to check?

Reply:
I checked that the implementation works fine with max and min integers as input data. But I found out other case which is unlikely to be passed. Could you check this one?
```
    def test_series_value_counts_uncovered_cases(self):
        def test_impl(S):
            return S.value_counts()

        # data from test_global_input_data_numeric
        data_to_test = [
            [1, -1, 0],
            [1., np.nan, -1., 0., min_float64, max_float64],
            [np.nan, np.inf, np.NINF, np.NZERO]
        ]

        hpat_func = hpat.jit(test_impl)

        for data in data_to_test:
            S = pd.Series(data)
            pd.testing.assert_series_equal(hpat_func(S), test_impl(S))
```